### PR TITLE
added walletloader to terminal package to prevent panic

### LINKED ISSUE #179: error/crash when running godcr if wallet doesn't exist
### Cli
Running a godcr command e.g. `godcr stakeinfo` without having a wallet, prompts user to create a wallet. If user choses not to create a wallet at the moment, godcr crashes.
screenshot 2019-01-25 at 2 04 57 pm

### Nuklear
The UI launch is canceled and the following error is printed after the app quits.`fatal error: unexpected signal during runtime execution`

### qt
UI launch is canceled without any message.

### terminal
If wallet doesn't exist, user is prompted to create wallet. If user does not accept to create wallet, the terminal UI is displayed even though the features in it cannot be used without a wallet. When the app is quit, the following message is displayed:
```
Blockchain sync failed to start. wallet_not_loaded
wallet_not_loaded
```

If user accepts to create a wallet, wallet is created. After wallet creation, user is asked if to sync blockchain. **Error is** blockchain sync happens whether user agrees to sync or not.

### PR DESCRIPTION
**Pull request by Sirmorrison**
fixes #179 for terminal

> I am experiencing a panic whenever I try to navigate to the `Stake info` page.
> I am using decrlibwallet, the `walletrpcserver` is not set in the config

implemented the terminal create wallet page and also added walletloader that fixes panic when using dcrlibwallet

### CODE REVIEW

**Review by ademuanthony — CHANGES_REQUESTED** [review]
Just a few changes

File: terminal/terminal.go [hunk 1]
```diff
@@ -6,24 +6,54 @@ import (
...
+				}
+				CreateWallet(ctx, password, walletMiddleware)
+			}).
+			AddButton("Quit", func() {
+				tviewApp.Stop()
+			})
+		form.SetBorder(true).SetTitle("Enter some data").SetTitleAlign(tview.AlignCenter).SetRect(30,10,40,10)
+		return tviewApp.SetRoot(form, false).Run()
+	
+	}else{
```
Comment (ademuanthony) [inline terminal/terminal.go]: you can invert the condition to do away with the else
```
 if walletExists {
    return tviewApp.SetRoot(layout, true).SetFocus(layout).Run()
}
````

File: terminal/walletloader.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			fmt.Fprintln(os.Stderr, err.Error())
+		}
+		return
+
+	case <-ctx.Done():
+		return false, ctx.Err()
+	}
+}
+
+func CreateWallet(ctx context.Context, password string, walletMiddleware app.WalletMiddleware) (err error) {
```
Comment (ademuanthony) [inline terminal/walletloader.go:52]: the `ctx` param is not used, we can do away with it

File: terminal/walletloader.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	// first check if wallet already exists
+	//work in progress
```
Comment (ademuanthony) [inline terminal/walletloader.go:54]: what is this for, I think it can go

Comment (Sirmorrison) [reply terminal/walletloader.go:54]: this is a future implementation for creating wallet will create a different pr to complete the implementation is only half implemented

Comment (Sirmorrison) [reply terminal/walletloader.go:54]: ctx will be used for blockchain sync function which will be implemented in future

**Review by ademuanthony — APPROVED** [review]

**Pull request merged**